File: Solution/0034.Find_First_and_Last_Position_of_Element_in_Sorted_Array/0034.Find_First_and_Last_Position_of_Element_in_Sorted_Array.py
class Solution:
    def searchRange(self, nums: List[int], target: int) -> List[int]:
        i,j=self.lower_bound(nums,target),self.upper_bound(nums,target)
        if i>j:
            return [-1,-1]
        else:
            return [i,j]
    def lower_bound(self,nums,target):
        i,j=0,len(nums)
        while i<j:
            m=i+(j-i)//2
            if nums[m]>=target: j=m
            else: i=m+1
        return i;
    def upper_bound(self,nums,target):
        i,j=0,len(nums)
        while i<j:
            m=i+(j-i)//2
            if nums[m]<=target: i=m+1
            else: j=m
        return j-1

File: Solution/0034.Find_First_and_Last_Position_of_Element_in_Sorted_Array/0034.Find_First_and_Last_Position_of_Element_in_Sorted_Array.cpp
class Solution {
public:
    vector<int> searchRange(vector<int>& nums, int target) {
        int i=lower_bound(nums,target);
        int j=upper_bound(nums,target);
        if(i>j) return {-1,-1};
        else return {i,j};
    }
    int lower_bound(vector<int>& nums,int target)
    {
        int i=0,j=nums.size();
        int m;
        while(i<j)
        {
            m=i+(j-i)/2;
            if(nums[m]>=target) j=m;
            else if(nums[m]<target) i=m+1;
        }
        return i;
    }
    int upper_bound(vector<int>& nums,int target)
    {
        int i=0,j=nums.size();
        int m;
        while(i<j)
        {
            m=i+(j-i)/2;
            if(nums[m]<=target) i=m+1;
            else if(nums[m]>target) j=m;
        }
        return j-1;
    }
};
